flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobMessageParameters.java

```diff
@@ -26,15 +26,16 @@
  */
 public class JobMessageParameters extends MessageParameters {
 
-	private final JobIDPathParameter jobPathParameter = new JobIDPathParameter();
+	private final JobIDPathParameter jobIDPathParameter = new JobIDPathParameter();
+	private final JobIDQueryParameter jobIDQueryParameter = new JobIDQueryParameter();
```

Comment:
I would suggest to instead create an abstract MetricsMessageParameters class (containing the get thingie) and extend that for job/task/subtask.